crates/autopilot/src/run_loop.rs

```diff
@@ -555,6 +555,24 @@ impl RunLoop {
             std::cmp::Reverse(participant.solution().score().get().0)
         });
 
+        // Filter out solutions that don't come from their corresponding submission
+        // address
+        let mut solutions = solutions
+            .into_iter()
+            .filter(|participant| {
+                let submission_address = participant.driver().submission_address;
+                let is_solution_from_driver = participant.solution().solver() == submission_address;
```

Comment:
this will cause issues in case copium wants to use multiple different solvers, cc @bram-vdberg